### PR TITLE
in game votes now provoke an action button

### PR DESCRIPTION
**Pull request by Incoming5643**
Even as a ghost or a new_player on the title screen.

Will need to check to make sure the generalizing of these procs doesn't break any other actions using owner typecast as a living. Seems unlikely but better safe than sorry.

d10_icon

### CODE REVIEW

**Comment by Zilenan91** [comment]
Cleanbot

**Comment by Bawhoppen** [comment]
Action buttons should be for IC stuff, maybe make a seperate spot on the top right side of the screen for this?

File: code/controllers/subsystem/voting.dm [hunk 1]
```diff
@@ -186,6 +189,10 @@ var/datum/subsystem/vote/SSvote
 		log_vote(text)
 		world << "\n<font color='purple'><b>[text]</b>\nType <b>vote</b> or click <a href='?src=\ref[src]'>here</a> to place your votes.\nYou have [config.vote_period/10] seconds to vote.</font>"
 		time_remaining = round(config.vote_period/10)
+		for(var/client/C in clients)
```
Comment (RemieRichards) [inline code/controllers/subsystem/voting.dm]: this should be an istypeless forloop.

File: code/controllers/subsystem/voting.dm [hunk 2]
```diff
@@ -271,6 +278,11 @@ var/datum/subsystem/vote/SSvote
 			submit_vote(round(text2num(href_list["vote"])))
 	usr.vote()
 
+/datum/subsystem/vote/proc/remove_action_buttons()
+	for(var/datum/action/vote/V in generated_actions)
```
Comment (RemieRichards) [inline code/controllers/subsystem/voting.dm]: this should be an istypeless forloop.

**Comment by Metacide** [comment]
:+1:

**Comment by Incoming5643** [comment]
@Bawhoppen There's a conflict of what that space is used for between the living and dead.

In the dead that slot is use exclusively to show ghost role slots
In the living that slot is used to show "terrible environmental problems that are killing you right now"

So I kinda don't want to show anything popping up in the upper right because in both cases it might mislead the players into thinking it's something else.

**Comment by JJRcop** [comment]
we gotta rethink our real estate

**Comment by optimumtact** [comment]
Remove the chat panel I say, it will improve the game

**Pull request merged**